timbermill-local/src/main/java/com/datorama/oss/timbermill/ElasticsearchClient.java

```diff
@@ -671,62 +678,54 @@ long countByName(String name, String env) throws IOException {
-		if (failureMessage != null && failureMessage.contains("type=null_pointer_exception")){
-			DbBulkRequest failedRequest = extractFailedRequestsFromBulk(dbBulkRequest, responses);
-			LOG.error("Null Pointer Exception Error in script. Requests:");
-			failedRequest.getRequest().requests().forEach(r -> LOG.error(r.toString()));
-	 	}
-	 	else {
-			if (retryNum >= numOfElasticSearchActionsTries) {
-				LOG.error("Reached maximum retries ({}) attempt to index. Failure message: {}", numOfElasticSearchActionsTries, failureMessage);
-				if (diskHandler != null) {
-					if (dbBulkRequest.getTimesFetched() < maxBulkIndexFetches) {
-						DbBulkRequest updatedDbBulkRequest = extractFailedRequestsFromBulk(dbBulkRequest, responses);
-						try {
-							diskHandler.persistBulkRequestToDisk(updatedDbBulkRequest);
-							if (dbBulkRequest.getTimesFetched() == 0) {
-								numOfBulksPersistedToDisk.incrementAndGet();
-							}
-						} catch (MaximumInsertTriesException e) {
-							LOG.error("Tasks of failed bulk will not be indexed because couldn't be persisted to disk for the maximum times ({}).", e.getMaximumTriesNumber());
-							numOfCouldNotBeInserted.incrementAndGet();
-						}
-					} else {
-						LOG.error("Tasks of failed bulk {} will not be indexed because it was fetched maximum times ({}).", dbBulkRequest.getId(), maxBulkIndexFetches);
-						numOfFetchedMaxTimes.incrementAndGet();
-					}
-				} else {
-					LOG.error("Tasks of failed bulk will not be indexed because it was fetched maximum times ({}).", maxBulkIndexFetches);
+	private void usePersistence(DbBulkRequest dbBulkRequest, BulkResponse responses) {
+		if (dbBulkRequest.getTimesFetched() < maxBulkIndexFetches) {
+			DbBulkRequest updatedDbBulkRequest = extractFailedRequestsFromBulk(dbBulkRequest, responses);
+			try {
+				diskHandler.persistBulkRequestToDisk(updatedDbBulkRequest);
+				if (dbBulkRequest.getTimesFetched() == 0) {
+					numOfBulksPersistedToDisk.incrementAndGet();
 				}
-			} else {
-				LOG.warn("Failed while trying to bulk index tasks, failure message: {}. Going to retry.", failureMessage);
-				DbBulkRequest updatedDbBulkRequest = extractFailedRequestsFromBulk(dbBulkRequest, responses);
-				failedRequests.offer(Pair.of(updatedDbBulkRequest, retryNum));
+			} catch (MaximumInsertTriesException e) {
+				LOG.error("Tasks of failed bulk will not be indexed because couldn't be persisted to disk for the maximum times ({}).", e.getMaximumTriesNumber());
+				numOfCouldNotBeInserted.incrementAndGet();
 			}
+		} else {
+			LOG.error("Tasks of failed bulk {} will not be indexed because it was fetched maximum times ({}).", dbBulkRequest.getId(), maxBulkIndexFetches);
+			numOfFetchedMaxTimes.incrementAndGet();
 		}
 	}
 
-	private DbBulkRequest extractFailedRequestsFromBulk(DbBulkRequest dbBulkRequest, BulkResponse bulkResponses) {
+	public DbBulkRequest extractFailedRequestsFromBulk(DbBulkRequest dbBulkRequest, BulkResponse bulkResponses) {
```

Comment:
changed to public in order to use it in a test class, 
better to duplicate it?

Reply:
you can use 'package private'

Reply:
+1